commit dc5382e1b765
Author: Lee Salzman <lsalzman@mozilla.com>
Date:   Fri May 25 00:57:45 2018 -0400

    Bug 1462682 - Skia path bounds rounding fix. r=rhunt, a=RyanVM
    
    MozReview-Commit-ID: Lm0XhyLLCCV
    
    --HG--
    extra : source : 784deba1907770c8f4c3482509ae99d474c4439f
---
 gfx/skia/skia/src/core/SkScan_Path.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git gfx/skia/skia/src/core/SkScan_Path.cpp gfx/skia/skia/src/core/SkScan_Path.cpp
index 2373e62d46ff2..90a22305260f1 100644
--- gfx/skia/skia/src/core/SkScan_Path.cpp
+++ gfx/skia/skia/src/core/SkScan_Path.cpp
@@ -564,7 +564,12 @@ static bool clip_to_limit(const SkRegion& orig, SkRegion* reduced) {
 // Bias used for conservative rounding of float rects to int rects, to nudge the irects a little
 // larger, so we don't "think" a path's bounds are inside a clip, when (due to numeric drift in
 // the scan-converter) we might walk beyond the predicted limits.
-static const double kConservativeRoundBias = 0.5 + 0.5 / SK_FDot6One;
+//
+// This value has been determined trial and error: pick the smallest value (after the 0.5) that
+// fixes any problematic cases (e.g. crbug.com/844457)
+// NOTE: cubics appear to be the main reason for needing this slop. If we could (perhaps) have a
+// more accurate walker for cubics, we may be able to reduce this fudge factor.
+static const double kConservativeRoundBias = 0.5 + 1.5 / SK_FDot6One;
 
 /**
  *  Round the value down. This is used to round the top and left of a rectangle,
